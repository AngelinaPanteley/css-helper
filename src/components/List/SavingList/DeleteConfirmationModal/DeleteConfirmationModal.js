import React from 'react';
import PropTypes from 'prop-types';
import styles from './DeleteConfirmationModal.scss';
import Modal from '../../../../hoc/Modal/Modal';

const deleteConfirmationModal = (props) => {
  return (
    <Modal isOpen={props.isOpen}
      onClose={props.onClose}>
      <p className={styles.ModalTitle}>
        Are you sure you want to delete this saving?
      </p>
      <div className={styles.ButtonContainer}>
        <button
          onClick={props.onClose}>
          Cancel
        </button>
        <button
          className={styles.Confirm}
          onClick={props.onConfirm}>
          Delete
        </button>
      </div>
    </Modal>
  );
}

export default deleteConfirmationModal;