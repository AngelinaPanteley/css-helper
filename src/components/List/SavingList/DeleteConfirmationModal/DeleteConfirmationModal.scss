@import '../../../../variables';

.ModalTitle {
  color: white;
  margin-bottom: 20px;
  font-size: 20px;
}

.ButtonContainer {
  @include flex-wrapper(row, flex-end);

  width: 100%;

  * {
    margin-left: 10px;
  }
  
  button {
    text-decoration: none;
    color: $color-white;
    transition: $transition;
    font-size: 15px;
    text-transform: uppercase;
    padding: 9px 30px 10px;
    border: 1px solid $color-white;
    background: rgba(56, 58, 71, 0.55);
    align-self: flex-end;
    cursor: pointer;
    outline: none;
  }

  button{
    &:hover,
    &:focus,
    &:active {
      background: rgba(255, 255, 255, 0.1);
      transition: $transition;
      color: $color-white;
    }
  }

  .Confirm {
    background: rgba(255, 0, 81, 0.4);

    &[disabled] {
      cursor: not-allowed;
    }

    &:hover,
    &:focus[disabled=false],
    &:active[disabled=false] {
      &:not([disabled]) {
        background: rgba(255, 0, 81, 0.6);
        transition: $transition;
        color: $color-white;
      }
    }
  }
}