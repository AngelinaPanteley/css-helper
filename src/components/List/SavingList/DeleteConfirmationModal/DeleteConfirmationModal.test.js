"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { expect as chaiExpect } from 'chai';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.1';
import DeleteConfirmationModal from './DeleteConfirmationModal.js';

configure({ adapter: new Adapter() });

describe('>>>DeleteConfirmationModal', () => {

  //SNAPSHOTS

  const props = {
    isOpen: true,
    onClose: () => { },
    onConfirm: () => { },
  };

  const modalComponent = <DeleteConfirmationModal {...props} />;
  const component = renderer.create(modalComponent);

  it('renders correctly', () => {
    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();
  });

  //SHALLOW RENDERING

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(modalComponent);
  });

  it('renders Modal', () => {
    chaiExpect(wrapper.find('Modal')).to.have.lengthOf(1);
  });

  it('renders two buttons', () => {
    chaiExpect(wrapper.find('button')).to.have.lengthOf(2);
  });
});
