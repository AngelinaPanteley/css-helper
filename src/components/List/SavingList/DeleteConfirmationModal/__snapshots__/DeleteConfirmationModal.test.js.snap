// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>DeleteConfirmationModal renders correctly 1`] = `
Array [
  <div
    className=" "
    onClick={[Function]}
  />,
  <div
    className=" "
  >
    <p>
      Are you sure you want to delete this saving?
    </p>
    <div>
      <button
        onClick={[Function]}
      >
        Cancel
      </button>
      <button
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
  </div>,
]
`;
